index.js

```diff
@@ -33,10 +33,15 @@ const logError = (id, errorType, err) => {
   }
 };
 
-const submitAudit = async (table, opts) => {
+const submitAudit = async (duplicate, opts) => {
   try {
     if (config.audit) {
-      await db(table).insert(opts);
+      if(duplicate) {
```

Comment:
This is a bit of an anti-pattern. Rather than pass booleans as arguments to a method to drive a separate process it is better to either pass the table name to the submitAudit method and call it twice.

Reply:
Yeah - understand this. At the moment it's the only way i've managed to avoid unhandled promise rejections and get the errors to be where we need them so that the message isnt deleted from the queue on failure.